packages/flutter/lib/src/foundation/serialization.dart

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:typed_data/typed_buffers.dart' show Uint8Buffer;
+
+/// Write-only buffer for incrementally building a [ByteData] instance.
+///
+/// A WriteBuffer instance can be used only once. Attempts to reuse will result
+/// in [NoSuchMethodError]s being thrown.
+///
+/// The byte order of serialized data is [Endianness.BIG_ENDIAN].
+/// The byte order of deserialized data is [Endianness.HOST_ENDIAN].
+class WriteBuffer {
+  Uint8Buffer _buffer;
+  ByteData _eightBytes;
+  Uint8List _eightBytesAsList;
+
+  WriteBuffer() {
```

Comment:
nit: here and in the class below, please move the constructor to above the fields. We put constructors first so that someone quickly glancing at the code can quickly tell if the class has an implicit constructor or explicit constructors.